packages/ckeditor5-link/src/ui/linkactionsview.ts

```diff
@@ -208,9 +211,34 @@ export default class LinkActionsView extends View {
 					'ck-link-actions__preview'
 				],
 				href: bind.to( 'href', href => href && ensureSafeUrl( href, this._linkConfig.allowedProtocols ) ),
-				target: '_blank',
-				rel: 'noopener noreferrer'
+				target: bind.to( 'href', href => {
+					if ( isScrollableToBookmark( href, this._bookmarkEditing ) ) {
+						return '_self';
+					}
+
+					return '_blank';
+				} ),
+				rel: bind.to( 'href', href => {
+					if ( isScrollableToBookmark( href, this._bookmarkEditing ) ) {
+						return 'noopener';
+					}
+
+					return 'noopener noreferrer';
+				} )
+			},
+			on: {
+				click: bind.to( () => {
```

Comment:
Maybe a `evt.preventDefault()` call would be good here so we could ignore other link related attributes for in editable scroll.